From e703bef8d53437804cea3d7e82e8322ecb4489ec Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Tue, 14 Aug 2018 16:32:19 +0200
Subject: [PATCH] Merge pull request #113 in SW_UGW/linux from
 bugfix/PMCPESYS-68-grx500-thermal-not-all-sensor-will-be-registered to master

* commit '697421001ad11c45ebd89fe8ce9437e029e3fa7c':
  PMCPESYS-68: thermal: ltq: Register sensor with index 0 as well
---
 drivers/thermal/ltq_thermal.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/thermal/ltq_thermal.c b/drivers/thermal/ltq_thermal.c
index 2438926c6545..f662ea6c82eb 100644
--- a/drivers/thermal/ltq_thermal.c
+++ b/drivers/thermal/ltq_thermal.c
@@ -260,11 +260,9 @@ static int ltq_thermal_probe(struct platform_device *pdev)
 		return PTR_ERR(priv->chiptop);
 	}
 
-	if (device_property_read_u32(&pdev->dev, "#thermal-sensor-cells",
-				     &priv->count)) {
-		/* register single cell if #thermal-sensor-cells is missing */
-		priv->count = 0;
-	}
+	device_property_read_u32(&pdev->dev, "#thermal-sensor-cells",
+				 &priv->count);
+	priv->count++;
 
 	priv->sensors = devm_kzalloc(&pdev->dev, priv->count *
 				     sizeof(struct ltq_thermal_sensor),
